zipline/algorithm.py

```diff
@@ -383,6 +384,40 @@ def __init__(self, *args, **kwargs):
 
         self.benchmark_sid = kwargs.pop('benchmark_sid', None)
 
+        self.capital_changes = kwargs.pop('capital_changes', {})
+        if self.capital_changes:
+            self.capital_changes = \
+                {pd.Timestamp(iso8601.parse_date(dt_str).
+                              strftime('%Y-%m-%d %H:%M'), tz='UTC'): v
+                 for dt_str, v in iteritems(self.capital_changes)}
+
+            for dt in copy(self.capital_changes):
```

Comment:
IMO, all of these preprocessing and mutating steps don't belong here.  If we need data in a specific format, it should be up to the client to provide it in that format.